protocols/imap/src/main/java/org/apache/james/imap/processor/fetch/FetchProcessor.java

```diff
@@ -201,10 +201,12 @@ private Mono<Void> doFetch(SelectedMailbox selected, FetchRequest request, Respo
         List<MessageRange> ranges = new ArrayList<>();
 
         for (IdRange range : request.getIdSet()) {
-            MessageRange messageSet = messageRange(session.getSelected(), range, request.isUseUids())
-                .orElseThrow(() -> new MessageRangeException(range.getFormattedString() + " is an invalid range"));
-            if (messageSet != null) {
-                MessageRange normalizedMessageSet = normalizeMessageRange(selected, messageSet);
+            if (session.getSelected().getLastUid().isEmpty()) {
+                continue;
+            }
```

Comment:
See https://github.com/apache/james-project/pull/2385#issuecomment-2302370425

I am not fan of basing a decision (skip the data) on a projection (the state of the local in-memory data structure "Selected Mailbox" may be temporarilry out-of-date especially in a distributed setup)

On the example provided the MSN id set conversion `1:*` bear the meaning *all messages* and it would make sense to me to convert it into `MessageRange.all()`. (right now we are translating `1:*` into `MessageRange.of(firstUid, MessageUid.MAX)`.

This would even make sens regarding performances as downstream code akka the mailbox is optimized to handle `MessageRange.all()`.

Thoughts?

Reply:
Yes this was.